core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManagerConfig.java

```diff
@@ -196,14 +180,107 @@ public class RemoteStorageManagerConfig extends AbstractConfig {
                 METRICS_RECORDING_LEVEL_DOC);
     }
 
+    /**
+     * Internal config for encryption.
+     *
+     * <p>It's needed for more convenient dynamic config definition.
+     */
+    private static class EncryptionConfig extends AbstractConfig {
+        private static final String ENCRYPTION_KEY_PAIR_ID_CONFIG = "encryption.key.pair.id";
+        private static final String ENCRYPTION_KEY_PAIR_ID_DOC =
+            "The ID of the key pair to be used for encryption";
+
+        private static final String ENCRYPTION_KEY_PAIRS_CONFIG = "encryption.key.pairs";
+        private static final String ENCRYPTION_KEY_PAIRS_DOC = "The list of encryption key pair IDs";
+
+        private static final String ENCRYPTION_PUBLIC_KEY_FILE_DOC = "The path to the RSA public key file";
+        private static final String ENCRYPTION_PRIVATE_KEY_FILE_DOC = "The path to the RSA private key file";
+
+        private EncryptionConfig(final ConfigDef configDef, final Map<String, ?> props) {
+            super(configDef, props);
+        }
+
+        Path encryptionPublicKeyFile(final String keyPairId) {
+            return Path.of(getString(publicKeyFileConfig(keyPairId)));
+        }
+
+        Path encryptionPrivateKeyFile(final String keyPairId) {
+            return Path.of(getString(privateKeyFileConfig(keyPairId)));
+        }
+
+        public static EncryptionConfig create(final Map<String, ?> props) {
+            final ConfigDef configDef = new ConfigDef();
+            // First, define the active key ID and key ID list fields, they are required always.
+            configDef.define(
+                ENCRYPTION_KEY_PAIR_ID_CONFIG,
+                ConfigDef.Type.STRING,
+                ConfigDef.NO_DEFAULT_VALUE,
+                ConfigDef.Importance.HIGH,
+                ENCRYPTION_KEY_PAIR_ID_DOC
+            );
+            configDef.define(
+                ENCRYPTION_KEY_PAIRS_CONFIG,
+                ConfigDef.Type.LIST,
+                ConfigDef.NO_DEFAULT_VALUE,
+                ConfigDef.Importance.HIGH,
+                ENCRYPTION_KEY_PAIRS_DOC
+            );
+            final EncryptionConfig encryptionConfig1 = new EncryptionConfig(configDef, props);
```

Comment:
nit:
```suggestion
            final EncryptionConfig config = new EncryptionConfig(configDef, props);
```

Reply:
Thanks, sure! I just first had `encryptionConfig2` too. Renamed to `interimEncryptionConfig`.